File: schoolWork/GenerateHTML.java
/*
 * Name: Shawn G
 * Date: 11/17/2015
 * Summary: This class will contain a method for creating the array and reading
 * into an HTML template, along with generating an HTML file.
 */

import java.io.File;
import java.io.PrintWriter;
import java.util.ArrayList;
import java.util.Scanner;

public class GenerateHTML {
    
    ArrayList<String> html = new ArrayList<>();
    
    public void CreateArray(){
        String path = "template.txt";  // Path to read the html template      
        try{
            Scanner read = new Scanner(new File(path));                                           // Scanner to read from the html file            
            
            while(read.hasNext() == true){                                                        // If there are still lines in the file, continue the loop
                html.add(read.nextLine());                                                        // Add line of text into html arraylist
            }
        } catch(Exception e){
            System.out.println("An error has occured in 'CreateArray': " + e);                    // Error catching
        }
    }
    public void linkFinalization(){
        ArrayList<String> htmlLink = new ArrayList<>();
        
        //System.out.println("Reached linkFinalization"); // Troubleshooting
        
        for(int i = 0; i < html.size(); i++){
            if(html.get(i).matches(".*?\\bhttp\\b.*?")){                        // Searches html arraylist for the words 'http'
                String[] splitHtml = html.get(i).split("<td>|<br />|</td>");    // Splits the table code out and puts the URL's into an array
                html.set(i, "");                                                // Clears out the old line of HTML code for replacement
                for(int r = 1; r < splitHtml.length; r++){
                    String url = "          <a href=\"" + splitHtml[r] + "\">" + splitHtml[r] + "</a><br />";  // Creates HTML code for links out of the URL string index
                    html.add(i, url);                                           // Inserts the url into the location of the code
                    //System.out.println(url); // Troubleshooting
                    i+= 1;                                                      // Keeps links in order of which they are put                    
                }
                break;                                                          // Break loop when we find and replace the code
            }
        }
    }
    public void generateHTML(){
        String outputPath = "assignments.html";  // File path for HTML document
        try{
            PrintWriter write = new PrintWriter(outputPath);                    // Used for writing into the HTML output file
            
            for(int i = 0; i < html.size(); i++){
                write.println(html.get(i));                                     // Creates and writes each line of text from html arraylist into the output file location
            }
            write.close();                                                      // Close document to save text
            System.out.println("HTML document generated at: " + outputPath);    // Message to let user know process completed
        } catch (Exception e){
            System.out.println("An error has occured in 'generateHTML': " + e); // Error catching
        }
    }
}

File: schoolWork/CSharp/AssignmentGenerator/GenerateHTML.cs
using System;
using System.Text.RegularExpressions;
using System.IO;
using System.Collections;

/// <summary>
/// Name: Shawn Giroux
/// Date: 11/19/2015
/// Summary: This class will contain a method that gets user input, converts it
/// into a line of HTML code, and outputs it to finalize our HTML document. It
/// will also return a boolean of 'true' to let the user know we have completed
/// the process.
/// </summary>

namespace AssignmentGenerator
{
    class GenerateHTML
    {

        public ArrayList html = new ArrayList();  // Arraylist will hold HTML template for manipulation

        public void CreateArray()
        {
            string path = @"E:\Programming\CSharp\programs\AssignmentGenerator\AssignmentGenerator\src\template.txt";  // Path to the template
            try
            {
                String[] htmlArray = File.ReadAllLines(path);   // Reads lines from the file into an array
                foreach(var item in htmlArray)                  // Throws the array into an arraylist so we can add elements later
                {
                    html.Add(item);
                }
            } catch(Exception e)
            {
                Console.WriteLine("An error has occured in 'CreateArray': " + e);  // Error catching
            }
        }
        public void linkFinalization()
        {
            Regex findURL = new Regex(@".\bhttp");            // Regular expression constructor for advanced keyword searches (http)

            for (int i = 0; i < html.Count; i++)              // For loop for searching through html
            {
                string checkHtml = html[i].ToString();        // String used for checking HTML
                if (findURL.IsMatch(checkHtml))               // If it finds a match to ReGex, do the following
                {
                    String[] splitHtml = Regex.Split(checkHtml, ("<td>|<br />|</td>"));  // Wipes the line of code of previous HTML coding and puts the leftover into an array
                    html[i] = "";                             // Clears the old line of code
                    foreach(var item in splitHtml)            // Goes through our new array to generate HTML links
                    {
                        if (item != "" && item.Contains(" ") != true)  // Avoids using the blanks slots in the array to get only what we need
                        {
                            String url = "          <a href=\"" + item + "\">" + item + "</a><br />";  // HTML link code
                            html.Insert(i, url);              // Inserts value into the location of our HTML arraylist
                            i += 1;                           // Allows our links to format in the way the user types them
                        }
                    }
                    break;                                    // Once we found our HTML code, we escape from the loop to avoid infinite looping
                }
            }
        }
        public void generateHTML()
        {
            string path = @"E:\Programming\CSharp\programs\AssignmentGenerator\AssignmentGenerator\src\assignment.html";  // Output file path
            string[] finalHTML = html.ToArray(typeof(string)) as string[];     // Convert out arraylist to a string array to allow us to use File.WriteAllLines
            try
            {
                File.WriteAllLines(path, finalHTML);                           // Prints into our output filepath
            }
            catch(Exception e)
            {
                Console.WriteLine("An error has occured in 'generateHTML': " + e); // Error catching
            }
        }
    }
}
